contrib/utilities/run_clang_tidy.sh

```diff
@@ -40,10 +40,11 @@ if test ! -d "$SRC/source" -o ! -d "$SRC/include" -o ! -d "$SRC/examples" -o ! -
 fi
 echo "SRC-DIR=$SRC"
 
+# do not allow bundled packages, otherwise we get too many warnings from TBB/UMFPACK/etc.
 # disable muparser (to avoid warnings with bundled one)
 # enable MPI (to get MPI warnings)
 # export compile commands (so that run-clang-tidy.py works)
-ARGS="-D DEAL_II_WITH_MUPARSER=OFF -D DEAL_II_WITH_MPI=ON -D CMAKE_EXPORT_COMPILE_COMMANDS=ON @$"
+ARGS="-D DEAL_II_ALLOW_BUNDLED=OFF -D DEAL_II_WITH_MUPARSER=OFF -D DEAL_II_WITH_MPI=ON -D CMAKE_EXPORT_COMPILE_COMMANDS=ON $@"
```

Comment:
I see no reason to disable `muParser` support anymore. 

I am not sure if we indeed want to require `MPI`. I am not starting from a clean build directory when running the script. Otherwise, I would not be able to run with a maximum of dependencies enabled.
In the end, I don't care too much. It's just a bit strange to treat `MPI` special.

Reply:
We check "mpi-*", so I think we need MPI enabled for that. I agree about muparser though.

Reply:
`clang-tidy´ would not complain about missing `MPI` support but the `mpi-*` checks (probably) only detect anything when `MPI` support is enabled. If we expect people different from the developers to run the script, e.g. for fixing their PRs, I think it is reasonable to drop the requirement. I don't feel strongly about this, though.

Reply:
You can always specify ``-D DEAL_II_WITH_MPI=OFF`` in the additional options...

Reply:
Fine with me. Let's see if that will come up again when we enforce the checks via CI.